### PR TITLE
Add: add java configuration example for mybatis (5841)

### LINKED ISSUE #5841: make example cover all posibility
### Is your feature request related to a problem?
no

### Describe the feature you would like.
Refactor the example module to cover all usage combinations

Followings are the combinations

<details>
  <summary>sharding-jdbc</summary>

- [ ] sharding
    - [ ] java
    - [ ] yaml
    - [ ] properties
    - [ ] springboot
    - [ ] spring-namespace
- [ ] master-slave
    - [ ] java
    - [ ] yaml
    - [ ] properties
    - [ ] springboot
    - [ ] spring-namespace
- [ ] encryption
    - [ ] java
    - [ ] yaml
    - [ ] properties
    - [ ] springboot
    - [ ] spring-namespace
- [ ] shadow
    - [ ] java
    - [ ] yaml
    - [ ] properties
    - [ ] springboot
    - [ ] spring-namespace
- [ ] replica
    - [ ] java
    - [ ] yaml
    - [ ] properties
    - [ ] springboot
    - [ ] spring-namespace
</details>
<details>
  <summary>proxy</summary>

- [ ] sharding
    - [ ] java
    - [ ] yaml
    - [ ] properties
    - [ ] springboot
    - [ ] spring-namespace
- [ ] master-slave
    - [ ] java
    - [ ] yaml
    - [ ] properties
    - [ ] springboot
    - [ ] spring-namespace
- [ ] encryption
    - [ ] java
    - [ ] yaml
    - [ ] properties
    - [ ] springboot
    - [ ] spring-namespace
- [ ] shadow
    - [ ] java
    - [ ] yaml
    - [ ] properties
    - [ ] springboot
    - [ ] spring-namespace
- [ ] replica
    - [ ] java
    - [ ] yaml
    - [ ] properties
    - [ ] springboot
    - [ ] spring-namespace
</details>

### PR DESCRIPTION
**Pull request by taojintianxia**
Fixes #5841.

Changes proposed in this pull request:
- add java configuration example for mybatis

### CODE REVIEW

File: examples/pom.xml [hunk 1]
```diff
@@ -56,6 +56,8 @@
     
         <btm.version>2.1.3</btm.version>
         <seata.version>1.0.0</seata.version>
+
+        <lombok.version>1.16.20</lombok.version>
```
Comment (terrymanu) [inline examples/pom.xml:60]: example do not need lombok, not all users know it

File: examples/shardingsphere-jdbc-example/sharding-example/sharding-java-configuration/mybatis-example/pom.xml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        <dependency>
+            <groupId>org.mybatis</groupId>
+            <artifactId>mybatis</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+        </dependency>
+    </dependencies>
+</project>
```
Comment (terrymanu) [inline examples/shardingsphere-jdbc-example/sharding-example/sharding-java-configuration/mybatis-example/pom.xml:44]: Need add eof

Comment (terrymanu) [inline examples/shardingsphere-jdbc-example/sharding-example/sharding-java-configuration/mybatis-example/pom.xml:44]: We need a blank line of the end of file

File: examples/shardingsphere-jdbc-example/sharding-example/sharding-java-configuration/mybatis-example/src/main/java/org/apache/shardingsphere/example/shardingjdbc/sharding/javaconfiguration/mybatis/ExampleMain.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    private static OrderRepository orderRepository;
+
+    public static void main(String... args) throws IOException, SQLException {
+        Reader reader = Resources.getResourceAsReader("mybatis-config.xml");
+        SqlSessionFactory sqlSessionFactory = new SqlSessionFactoryBuilder().build(reader);
+        SqlSession sqlSession = sqlSessionFactory.openSession();
+        orderRepository = sqlSession.getMapper(OrderRepository.class);
+
+        // init the environment
```
Comment (terrymanu) [inline examples/shardingsphere-jdbc-example/sharding-example/sharding-java-configuration/mybatis-example/src/main/java/org/apache/shardingsphere/example/shardingjdbc/sharding/javaconfiguration/mybatis/ExampleMain.java:45]: Please remove useless inline comment, method name can explain the mean

File: examples/shardingsphere-jdbc-example/sharding-example/sharding-java-configuration/mybatis-example/src/main/java/org/apache/shardingsphere/example/shardingjdbc/sharding/javaconfiguration/mybatis/ExampleMain.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    private static final int DATA_COUNT = 10;
+
+    private static OrderRepository orderRepository;
+
+    public static void main(String... args) throws IOException, SQLException {
+        Reader reader = Resources.getResourceAsReader("mybatis-config.xml");
+        SqlSessionFactory sqlSessionFactory = new SqlSessionFactoryBuilder().build(reader);
+        SqlSession sqlSession = sqlSessionFactory.openSession();
+        orderRepository = sqlSession.getMapper(OrderRepository.class);
+
```
Comment (terrymanu) [inline examples/shardingsphere-jdbc-example/sharding-example/sharding-java-configuration/mybatis-example/src/main/java/org/apache/shardingsphere/example/shardingjdbc/sharding/javaconfiguration/mybatis/ExampleMain.java:44]: please remove useless inline empty lines

**Comment by taojintianxia** [comment]
it's not necessary to add mybatis demo with Java configuration

**Pull request closed without merging**